src/tabmat/dense_matrix.py

```diff
@@ -164,8 +164,10 @@ def _cross_sandwich(
         raise TypeError
 
     def _get_col_stds(self, weights: np.ndarray, col_means: np.ndarray) -> np.ndarray:
-        """Get standard deviations of columns."""
-        sqrt_arg = transpose_square_dot_weights(self._array, weights) - col_means**2
+        """Get standard deviations of columns using weights `weights`."""
+        sqrt_arg = transpose_square_dot_weights(
+            self._array - col_means[np.newaxis, :], weights
```

Comment:
Can we do this within the loop in `transpose_square_dot_weights` so that we can avoid (temporarily) allocating a new array that's as large as `self._array`?

Reply:
It's a private function that we don't use anywhere else I believe, so also feel free to change its arguments (i.e., pass `col_means` to it) if that feels most natural.